tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -16,100 +16,202 @@
 import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.{Constructor, ConstructorException}
+import org.yaml.snakeyaml.representer.Representer
 
 import org.apache.spark.internal.Logging
 
 /**
- * A wrapper class that stores all the properties that would be recommended by the Auto-tuner.
- * Separate getter and setter methods are specified for each property for ease of access.
+ * A wrapper class that stores all the GPU properties.
+ * The BeanProperty enables loading and parsing the YAML formatted content using the
+ * Constructor SnakeYaml approach.
  */
-class Config {
-  private val properties: collection.mutable.Map[String, String] = collection.mutable.Map(
-    "spark.executor.instances" -> null,
-    "spark.rapids.sql.enabled" -> null,
-    "spark.executor.cores" -> null,
-    "spark.executor.memory" -> null,
-    "spark.rapids.sql.concurrentGpuTasks" -> null,
-    "spark.task.resource.gpu.amount" -> null,
-    "spark.sql.shuffle.partitions" -> null,
-    "spark.sql.files.maxPartitionBytes" -> null,
-    "spark.rapids.memory.pinnedPool.size" -> null,
-    "spark.executor.memoryOverhead" -> null
-  )
-
-  def getExecutorInstances: Int = properties("spark.executor.instances").toInt
-
-  def getExecutorCores: Int = properties("spark.executor.cores").toInt
-
-  def getExecutorMemory: String = properties("spark.executor.memory")
-
-  def getConcurrentGpuTasks: Int = properties("spark.rapids.sql.concurrentGpuTasks").toInt
-
-  def getTaskResourceGpu: Double = properties("spark.task.resource.gpu.amount").toDouble
-
-  def getShufflePartitions: Int = properties("spark.sql.shuffle.partitions").toInt
-
-  def getMaxPartitionBytes: String = properties("spark.sql.files.maxPartitionBytes")
-
-  def getPinnedPoolSize: String = properties("spark.rapids.memory.pinnedPool.size")
-
-  def getExecutorMemoryOverhead: String = properties("spark.executor.memoryOverhead")
+class GpuWorkerProps(
+    @BeanProperty var memory: String,
+    @BeanProperty var count: Int,
+    @BeanProperty var name: String) {
+  def this() {
+    this("0m", 0, "None")
+  }
+  def isMissingInfo: Boolean = {
+    count == 0 || memory.startsWith("0") || name == "None"
+  }
+  def isEmpty: Boolean = {
+    count == 0 && memory.startsWith("0") && name == "None"
+  }
+  /**
+   * If the GPU count is missing, it will set a default value based on the number of cores in the
+   * system. Adds GPU count for each 16 CPU core.
+   *
+   * @return true if the value has been updated.
+   */
+  def setDefaultGpuCountIfMissing(): Boolean = {
+    if (count == 0) {
+      count = AutoTuner.DEF_WORKER_GPU_COUNT
+      true
+    } else {
+      false
+    }
+  }
+  def setDefaultGpuNameIfMissing(): Boolean = {
+    if (name == "None") {
+      name = AutoTuner.DEF_WORKER_GPU_NAME
+      true
+    } else {
+      false
+    }
+  }
+  def setDefaultGpuMemIfMissing(): Boolean = {
+    if (memory.startsWith("0")) {
+      memory = AutoTuner.DEF_WORKER_GPU_MEMORY_MB.getOrElse(getName, "0m")
```

Comment:
It pulls the default from the static hashmap `AutoTuner.DEF_WORKER_GPU_MEMORY_MB`.
If the GPU device is not defined there, it will be "0m".
I changed that to force it to 16G if the default cannot be loaded from the static map.